applications/CoSimulationApplication/tests/test_rigid_body_solver.py

```diff
@@ -0,0 +1,2047 @@
+        Parameters = KM.Parameters('''{
+            "problem_data": {
+                "problem_name": "RigidBodyStandalone",
+                "start_time": 0.0,
+                "end_time": 1.0
+            },
+            "solver_settings": {
+                "time_integration_parameters": {
+                    "time_step": 0.01
+                },
+                "model_import_settings": {
+                    "input_type": "non"
+                }
+            },
+            "output_processes": [
+            ],
+            "processes": {
+            }
+        }''')
+        
+        self.assertRaises(Exception, input_check._CheckMandatoryInputParameters, Parameters)
+
+        
+    def test_CheckMandatoryInputParameters11(self):
+        # Missing restart_load_file_label in model_import_settings
+        Parameters = KM.Parameters('''{
+            "problem_data": {
+                "problem_name": "RigidBodyStandalone",
+                "start_time": 0.0,
+                "end_time": 1.0
+            },
+            "solver_settings": {
+                "time_integration_parameters": {
+                    "time_step": 0.01
+                },
+                "model_import_settings": {
+                    "input_type": "rest"
+                }
+            },
+            "output_processes": [
+            ],
+            "processes": {
+            }
+        }''')
+        
+        self.assertRaises(Exception, input_check._CheckMandatoryInputParameters, Parameters)
+
+
+    def test_ValidateAndAssignRigidBodySolverDefaults1(self):
+        # Check returned solver_settings
+        Parameters = KM.Parameters('''{
+            "problem_data": {
+            },
+            "solver_settings": {
+                "time_integration_parameters": {
+                    "time_step": 0.01
+                },
+                "model_import_settings": {
+                    "input_type": "none"
+                }
+            },
+            "output_processes": [
+            ],
+            "processes": {
+            }
+        }''')
+        solver_settings = Parameters["solver_settings"]
+        checked_solver_settings = input_check._ValidateAndAssignRigidBodySolverDefaults(solver_settings)
+        ref_solver_settings = KM.Parameters('''{
+            "active_dofs": [],
+            "buffer_size": 3,
+            "domain_size": 3,
+            "echo_level": 0,
+            "model_import_settings": {
+                "input_filename": "Main",
+                "input_output_path": "restart",
+                "input_type": "none",
+                "load_restart_files_from_folder": true,
+                "restart_load_file_label": "0.0"
+            },
+            "time_integration_parameters": {
+                "rho_inf": 0.16,
+                "time_step": 0.01
+            }
+        }''')
+        
+        self.assertEqual(ref_solver_settings.PrettyPrintJsonString(), checked_solver_settings.PrettyPrintJsonString())
```

Comment:
Usually we don't have to test for Validate and Assign, because on the long run, you may add new settings which does not change the old ones therefore extending features. Which will breeak this test. Even if you want to check, then you can use the following 

```python
self.assertTrue(ref_solver_settings.IsEquivalentTo(checked_solver_settings))
```

Reply:
don't use `PrettyPringJsonString()`

Reply:
I replaced PrettyPringJsonString() with IsEquivalentTo() for now. Should I remove this test instead?